docs/concepts/internationalization.md

```diff
@@ -0,0 +1,72 @@
+Internationalization and localization
+=====================================
+
+Internationalization, i18n for short, is a way of designing code so that it requires no changes to be adapted for various languages. Localization is the process of adapting displayed text to the user's language.
+
+I18n is implemented using _translation keys_. A translation key is a string that identifies a piece of displayable text in no specific language. For example, `tile.dirt.name` is the language key referring to the name of the Dirt block. This way, displayable text may be referenced with no concern for a specific language. The code requires no changes to be adapted in a new language.
+
+Localization will happen in the game's locale. In a Minecraft client the locale is specified by the language settings. On a dedicated server, the only supported locale is en_US. A list of available locales can be found on the [Minecraft Wiki](https://minecraft.gamepedia.com/Language#Available_languages).
+
+Language files
+--------------
+
+Language files are located by `assets/domain/lang/[locale].lang` (e.g. the US English translation for `examplemod` would be `assets/examplemod/lang/en_us.lang`). Resource pack format 3 requires the locale name to be lowercased. The file format is simply lines of key-value pairs separated by `=`. Lines starting with `#` are ignored. Lines without a separator are ignored. The file must be encoded in UTF-8.
+
+```properties
+# items
+item.examplemod.example_item.name=Example Item Name
+
+# blocks
+tile.examplemod.example_block.name=Example Block Name
+
+# commands
+commands.examplemod.examplecommand.usage=/example <value>
+```
+
+Including the comment `#PARSE_ESCAPES` anywhere in the file will enable the slightly different and more complex Java properties file format. This format is required to support multiline values.
+
+!!! note
+    Enabling `#PARSE_ESCAPES` enables changes various aspects of the parsing. Among other things, the `:` character will be treated as a key-value separator. It must either be excluded from translation keys or escaped using `\:`.
+
+Unlocalized names
+-----------------
+
+Blocks, Items and other Minecraft constructs have unlocalized names. Unlocalized names are simply translation keys that are used when displaying their names.
```

Comment:
Translation keys are kinda explained in the first section, 2nd paragraph. Maybe something like "In Minecraft (MCP) the translation keys are called `unlocalised names`"? or even merging the 2 paragraphs

Reply:
I have described an "unlocalized name" as a translation key for the names of `Blocks, Items and other Minecraft constructs` only, and not as a term to be used interchangably with "translation key". In 1.13 the term "unlocalized name" will be gone.

Some might disagree on this, so I need some people's opinions.

Reply:
> and not as a term to be used interchangably with "translation key"

True, not all translation keys are unlocalised names (it was early when I wrote that). It's still a little weird to be mentioning them as a different thing just because they're used in one (type of) place. `and other Minecraft constructs` is also super vague.

Perhaps it needs to mention the `.name` addition some things get?

> In 1.13 the term "unlocalized name" will be gone.

IMO that's worth mentioning in the doc given how close they are to release